Mage.Common/src/main/java/mage/remote/handlers/server/HeartbeatHandler.java

```diff
@@ -0,0 +1,76 @@
+package mage.remote.handlers.server;
+
+import io.netty.channel.ChannelHandlerAdapter;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.timeout.IdleState;
+import io.netty.handler.timeout.IdleStateEvent;
+import java.util.concurrent.TimeUnit;
+import mage.remote.DisconnectReason;
+import org.apache.log4j.Logger;
+import mage.remote.handlers.WriteListener;
+import mage.remote.interfaces.MageServer;
+import mage.remote.messages.PingMessage;
+import mage.remote.messages.PongMessage;
+
+/**
+ *
+ * @author BetaSteward
+ */
```

Comment:
This and all of the new classes need documentation and description of what they're supposed to do.

There's too many classes and layers for them to not have any